### ISSUE TITLE
Getting node IP address by object ref

### ISSUE THREAD

**Issue by YarShev** [ray-project/ray#13630]
### Describe your feature request

There is the following API to get object table: `ray.objects()`. However, object location is some node ID. It would be awesome to get node IP address as well.

**Comment by wuisawesome** [comment]
Btw, if anyone is looking for a short term fix for this, the mapping between NodeID and ip address can be found in `ray.nodes()` (between `NodeID` and `NodeManagerAddress`)

**Comment by YarShev** [comment]
@wuisawesome , thanks a lot for the excellent note! By the way, why is `ray.nodes()` for debugging only (by following docstring for the method)?

**Comment by joseph-x-li** [comment]
> @wuisawesome , thanks a lot for the excellent note! By the way, why is `ray.nodes()` for debugging only (by following docstring for the method)?

Reading through the codebase a little, I can think of three possible reasons for the designation that ray.nodes() be for debugging only.

1) ray.nodes() is untested/unmaintained and not meant to be called since it may break things in the background or other horrible stuff.
  - ray.nodes() is just an alias for ray.state.GlobalState.node_table(). node_table() itself is called by many client-facing functions like ray.cluster_resources(), so ray.nodes() should be safe-ish.

2) It returns a dictionary with an unstable interface.
  - This one I can believe a bit more. There are many parameters in the dict returned by node_table that could possibly change due to backend infrastructure modifications. In fact, we have an interesting piece of code on line 243:
`node_info["alive"] = node_info["Alive"]`. Not sure which capitalization we want but alright.

3) ray.nodes() might be renamed in the future, or the mechanism to access cluster state might be overhauled (possibly removing the need for such a function?). 
  - Why might we think this?  Interestingly, looking back through the commit history, we might be able to infer why the "debugging only" message was added. Taking a look at the specific [PR](https://github.com/ray-project/ray/pull/10325) where it happened, ray.nodes() was deprecated for the release of ray v1.0.0, as noted in this [google sheet](https://docs.google.com/spreadsheets/d/1F6VZ1R_fgDzpRZjXI5_6gaUwW2CQKi431dCDTR-Vm8s/edit#gid=0). 

ray.nodes() has survived into ray v2.0.0 dev, so for now, I am going to use ray.nodes() regardless of what docstring says.

**Comment by scottsun94** [comment]
cc: @rkooo567 @rickyyx 
The underlying need should be addressed by state api? https://docs.ray.io/en/latest/ray-observability/state/state-api.html

**Comment by rickyyx** [comment]
Yes, check this [output schema](https://docs.ray.io/en/latest/ray-observability/state/ray-state-api-reference.html#state-api-schema-obj). Let's know if this works.

**Comment by scottsun94** [comment]
@rickyyx The `ip` in the schema is the node ip?

@YarShev Does it solve your issue?

**Comment by rkooo567** [comment]
Btw the IP from the ray list objects is the IP of the owner, not the object.

**Comment by rkooo567** [comment]
so I don't think it can satisfy the requirement. We may need new APIs to expose the state of plasma store

**Comment by YarShev** [comment]
> @YarShev Does it solve your issue?

It would be nice to have a Ray python API using which it is possible to know which node the object is located on.

**Comment by rkooo567** [comment]
actually it is possible now. You can try 

```
ref = ray.put(obj)
ray.experimental.get_object_locations([ref])
```

**Comment by YarShev** [comment]
```python
import ray

ray.init()

ref = ray.put(1)
ray.experimental.get_object_locations([ref])
{
    ObjectRef(00ffffffffffffffffffffffffffffffffffffff0100000001000000):
    {
        'node_ids':  ['aded04230653a0179dab919bbef9026e34a0e78f4a3208f1a4e1424c'], 'object_size': 15
    }
}
```

As far as I understand `node_ids` mathes `NodeID` in ray.nodes(). If so, I think it is okay. I wonder why the `object_size` is 15 even just for `int`?

**Comment by YarShev** [comment]
Also, when the feature is going to be non-experimental?

**Comment by rkooo567** [comment]
I think that includes some metadata that's appended to all ray objects. I think we can promote this to public API even now. We need to just go through some process...

**Comment by rkooo567** [comment]
Let me try this by ray 2.2

**Comment by YarShev** [comment]
Sounds good!

**Comment by rkooo567** [comment]
sorry that this was slipped. Will fix by 2.3
